src/libpriv/rpmostree-core.cxx

```diff
@@ -1973,6 +1976,45 @@ rpmostree_context_prepare (RpmOstreeContext *self, GCancellable *cancellable, GE
         }
     }
 
+  /* And remote replacement packages */
+  for (auto &override_replace : packages_override_replace)
+    {
+      switch (override_replace.from_kind)
+        {
+        case rpmostreecxx::OverrideReplacementType::Repo:
+          {
+            const char *repo = override_replace.from.c_str ();
+            for (auto &pkg : override_replace.packages)
+              {
+                g_auto (HySubject) subject = hy_subject_create (pkg.c_str ());
+                HyNevra nevra = NULL;
+                // We don't support provides or filenames: we need a concrete package name that
+                // matches the base. Nevras or partial nevras (e.g. foobar.x86_64 or foobar-1.2) are
+                // supported.
+                hy_autoquery HyQuery query = hy_subject_get_best_solution (
+                    subject, sack, NULL, &nevra, FALSE, TRUE, FALSE, FALSE, FALSE);
+                auto pkgname = nevra->getName();
+                // this should never happen, but just in case
```

Comment:
Wait, isn't it an expected situation where a user typos a package name?  Should we dedup the error message with the "package not found" error?

Reply:
The `getName()` here checks that the string in the config could be parsed as something that includes a package name. So e.g. `foobar`, `foobar-1.2`, `foobar.x86_64` would all return `foobar` for `getName()`.